communication/src/main/java/org/jboss/da/communication/pom/PomAnalyzerImpl.java

```diff
@@ -186,6 +201,59 @@ private GAV generateGAV(ProjectVersionRef ref) {
         return new GAV(ref.getGroupId(), ref.getArtifactId(), ref.getVersionString());
     }
 
+    public List<MavenPomView> getGitPomView(String scmUrl, String revision, String pomPath,
+            List<String> repositories) throws PomAnalysisException {
+        File clonedDir = null;
+        File tempDir = null;
+        try {
+            clonedDir = scmManager.cloneRepository(SCMType.GIT, scmUrl, revision);
+
+            tempDir = Files.createTempDirectory("deps").toFile();
+
+            List<File> pomFiles = findAllPomFiles(clonedDir);
+
+            Map<File, ProjectVersionRef> projectVersionRefs = getProjectVersionRefs(tempDir,
+                    pomFiles);
+            List<MavenPomView> poms = new ArrayList<MavenPomView>();
+            MavenPomReader mavenPomReader = carto.getGalley().getPomReader();
+            for (Map.Entry<File, ProjectVersionRef> entry : projectVersionRefs.entrySet()) {
+                poms.add(mavenPomReader.read(entry.getValue(),
+                        getRepoLocations(tempDir, clonedDir, repositories)));
+            }
+
+            return poms;
+        } catch (ScmException | IOException | GalleyMavenException e) {
+            log.warn(e.getMessage());
+        } finally {
+            FileUtils.deleteQuietly(clonedDir);
```

Comment:
SCM class take care of clonedDir. If you delete it manually, you can compromise cache.